extensions-contrib/ddsketch/src/main/java/org/apache/druid/query/aggregation/ddsketch/DDSketchAggregatorFactory.java

```diff
@@ -0,0 +1,327 @@
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    return new DDSketchAggregator(metricFactory.makeColumnValueSelector(fieldName), relativeError, numBins);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    return new DDSketchBufferAggregator(metricFactory.makeColumnValueSelector(fieldName), relativeError, numBins);
+  }
+
+  public static final Comparator<DDSketch> COMPARATOR = Comparator.nullsFirst(
+      Comparator.comparingLong(a -> a.serializedSize())
+  );
+
+  @Override
+  public Comparator getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(@Nullable Object lhs, @Nullable Object rhs)
+  {
+    if (lhs == null) {
+      return rhs;
+    }
+    if (rhs == null) {
+      return lhs;
+    }
+    DDSketch union = (DDSketch) lhs;
+    union.mergeWith((DDSketch) rhs);
+    return union;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new DDSketchAggregatorFactory(name, name, relativeError, numBins);
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(
+        new DDSketchAggregatorFactory(
+            name,
+            fieldName,
+            relativeError,
+            numBins
+        )
+    );
+  }
+
+  @Override
+  public Object deserialize(Object serializedSketch)
+  {
+    return DDSketchUtils.deserialize(serializedSketch);
+  }
+
+  @Nullable
+  @Override
+  public Object finalizeComputation(@Nullable Object object)
+  {
+    return object == null ? null : ((DDSketch) object).getCount();
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty
+  public double getRelativeError()
+  {
+    return relativeError;
+  }
+
+  @JsonProperty
+  public int getNumBins()
+  {
+    return numBins;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  /**
+   * actual type is {@link DDSketch}
+   */
+  @Override
+  public ColumnType getIntermediateType()
+  {
+    return TYPE;
+  }
+
+  @Override
+  public ColumnType getResultType()
+  {
+    return TYPE;
+  }
+
+  // The bounded lower collapsing yields a size of roughly numBins * 8 in terms of size.
+  // Since negative and positive value stores exist per sketch, we multiply this by 2.
+  // then we add 1k for measure.
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return numBins * 8 * 2 + 1000;
```

Comment:
legitimately unsure what to do about this one. Can someone help?

Reply:
You can use `IntMath.checkedMultiply(numBins, 8 * 2 * 1000)`. However, can you confirm that in which case it can over or underflow (check using Show Paths), and perhaps add a relevant comment that why it doesn't over/underflow?

Reply:
I updated the comment and tried to label everything